### ISSUE TITLE
quickstart: escape sequence has been displayed with MacPorts' python

### ISSUE THREAD

**Issue by tk0miya** [sphinx-doc/sphinx#5335]
### Problem
- quickstart: escape sequence has been displayed with MacPorts' python
- reported by @cocoatomo 

```
(venv) [tomohiko@Luna prompt]$ sphinx-quickstart 
Welcome to the Sphinx 1.7.6 quickstart utility.
Please enter values for the following settings (just press Enter to
accept a default value, if one is given in brackets).
Selected root path: .
You have two options for placing the build directory for Sphinx output.
Either, you use a directory "_build" within the root path, or you separate
"source" and "build" directories within the root path.
[35m> Separate source and build directories (y/n) [n]: [39;49;00m
[Interrupted.]
```

#### Procedure to reproduce the problem
Invoke `sphinx-quickstart` simply.

#### Expected results
All questions are colored.

### Environment info
- OS: Mac
- Python version: 3.7.0 (installed with MacPorts)
- Sphinx version: 1.7.6

**Comment by tk0miya** [comment]
It seems this trouble is caused by libedit which is used by MacPorts by default.
```
(venv) [tomohiko@Luna prompt]$ python3.7
Python 3.7.0 (default, Jun 28 2018, 05:55:06) 
[Clang 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import readline
>>> input('\x01\x1b[35m\x02Hello world\x01\x1b[39;49;00m\x02')
[35mHello world[39;49;00m
''
>>> readline.__doc__
'Importing this module enables command line editing using libedit readline.'
>>> exit()
```

After installing `py37-gnureadline` package, it is resolved.
```
[tomohiko@Luna prompt]$ sudo port install py37-gnureadline
(snip)
[tomohiko@Luna prompt]$ sphinx-quickstart 
Welcome to the Sphinx 1.7.7 quickstart utility.

Please enter values for the following settings (just press Enter to
accept a default value, if one is given in brackets).

Selected root path: .

You have two options for placing the build directory for Sphinx output.
Either, you use a directory "_build" within the root path, or you separate
"source" and "build" directories within the root path.
> Separate source and build directories (y/n) [n]: 
[Interrupted.]
```

**Comment by tk0miya** [comment]
Fixed by #5336.
@cocoatomo Thank you for reporting!
